### DISCUSSION TITLE
How to deal with errors and validation when useFieldArray is optional but all his key is required

### DISCUSSION THREAD

**Discussion by Steffi3rd** [react-hook-form/react-hook-form#3626] [Q&A]
Hello ! 

TLDR : I have an useFieldArray with 3 fields.
  
`firstName: '',  lastName: '',  email: ''` form should be valid 
`firstName: 'x',  lastName: '',  email: ''` lastName, email are required 
`firstName: '',  lastName: 'x',  email: ''` firstName, email are required  
`firstName: '',  lastName: '',  email: 'a@b.cd'` firstName, lastName are required 
`firstName: 'x',  lastName: 'x',  email: '`' email are required
`firstName: 'x',  lastName: 'x',  email: 'a@b.cd'` form should be valid

I have a form with a fieldArray which is optional if all his key (`firstName`, `lastName`, `email`) are empty. Form should be valid. If one of the key is not empty, all other key of this field array is required. 

So I started with this :

```
defaultValues: {
   users: []
}
```

But with this, I have typescript errors 
`errors?.users?.[index]?.email` : Property 'email' does not exist on type 'FieldError'

Because, I think it's because `email` is not defined in `defaultValues`. 

So, I continue with this : 

```
defaultValues: {
   users: [{ firstName: '', lastName: '', email: '' }]
}
```

With this, no typescript errors ✅ and all keys are required ✅.
But we cannot submit the form until all key are filled. Form is considered not valid 😞  **That's problem I want to solve.**

I would like to be able to submit form with the empty field array (where `firstName`, `lastName`, `email` are empty).

Is there a way to do that please ?

**Comment by bluebill1049** [comment]
i think it's probably easier to use schema or build a custom resolver for this requirement.
